xclim/core/formatting.py

```diff
@@ -142,8 +143,17 @@ def parse_doc(doc: str) -> Dict[str, str]:
     return out
 
 
+annotation_patt = re.compile(r".*\{(.*)\}.*")
```

Comment:
Define inside function if not used elsewhere. rename `type_set_pattern` ?

Reply:
I'm defining it outside because that's the point of `compile` : do it only once. However, the performance gain is negligeable, might juste be more elegant to do it inside.